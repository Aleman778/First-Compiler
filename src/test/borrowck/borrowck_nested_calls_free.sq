// ignore

fn mutate(v: &mut i32) -> i32 {
    *v = 22;
    *v
}


fn add(v: &i32, w: i32) -> i32 {
    *v + w
}


fn main() {
    let mut a: i32 = 1;
    add(&a, mutate(&mut a));
}
