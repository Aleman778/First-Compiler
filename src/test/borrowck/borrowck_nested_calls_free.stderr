error: cannot borrow `a` as mutable because it is also borrowed as immutable
  --> src/test/borrowck/borrowck_nested_calls_free.sq:16:20
   |
LL |     add(&a, mutate(&mut a));
   |     --- --         ^^^^^^ mutable borrow occurs here
   |     |   |
   |     |   immutable borrow occurs here
   |     immutable borrow later used by call